docs/source/corev_hw_loop.rst

```diff
@@ -21,98 +21,105 @@ CORE-V Hardware Loop Extensions
+
+In order to catch **as early as possible** those software exceptions when executing a program either
+on a verification Reference Model or on a virtual platform Instruction Set Simulator, ``those model/simulation platforms
+must generate a fatal error`` with a meaningfull message related to Hardware Loops constraints violation.
+
 The HWLoop constraints are:
 
--  Start and End address of an HWLoop must be word aligned
+-  Start and End address of an HWLoop must be 32-bit aligned.
+
+-  End address of an HWLoop must point to the instruction just after the last one of the loop body.
 
 -  HWLoop body must contain at least 3 instructions.
-   An illegal exception is raised otherwise.
+
+-  When both loops are nested, the End address of the outermost HWLoop (must be #1) must be at least 2
+   instructions further than the End address of the innermost HWLoop (must be #0),
+   i.e. HWLoop[1].endaddress >= HWLoop[0].endaddress + 8.
 
 -  No Compressed instructions (RVC) allowed in the HWLoop body.
-   An illegal exception is raised otherwise.
 
--  No uncoditional jump instructions allowed in the HWLoop body.
-   An illegal exception is raised otherwise.
+-  No jump or branch instructions allowed in the HWLoop body.
 
--  No coditional branch instructions allowed in the HWLoop body.
-   An illegal exception is raised otherwise.
+-  No memory ordering instructions (fence, fence.i) allowed in the HWLoop body.
 
 -  No privileged instructions (mret, dret, ecall, wfi) allowed in the HWLoop body, except for ebreak.
-   An illegal exception is raised otherwise.
 
--  No memory ordering instructions (fence, fence.i) allowed in the HWLoop body.
-   An illegal exception is raised otherwise.
+The rationale of NOT generating any hardware exception when violating any of those constraints is that it would add resources
+(32-bit adders and substractors needed for the third and fourth rules) which are costly in area and power consumption.
+These additional (and costly) resources would be present just to catch situations that should never happen. 
+This in an architectural choice in order to keep CV32E40P area and power consumption to its lowest level.
 
--  The End address of the outermost HWLoop (#1) must be at least 2
-   instructions further than the End address innermost HWLoop (#0),
-   i.e. HWLoop[1].endaddress >= HWLoop[0].endaddress + 8
-   An illegal exception is raised otherwise.
+The rationale of putting the end-of-loop label to the first instruction after the last one of the loop body
+is that it greatly simplifies compiler optimization (relative to basic blocks management).
 
-In order to use hardware loops, the compiler needs to setup the loop
-beforehand with the following instructions. Note that the minimum loop
-size is 3 instructions and the last instruction cannot be any jump or
-branch instruction.
+In order to use hardware loops, the compiler needs to setup the loops beforehand with the 6 defined instructions.
```

Comment:
In order to use hardware loops, the compiler needs to setup the loops beforehand with the cv.start/i, cv.count/i, and cv.stop/i instructions.